airbyte-integrations/connectors/source-metabase/source_metabase/streams.py

```diff
@@ -53,9 +62,100 @@ def path(self, **kwargs) -> str:
         return "dashboard"
 
 
-class Users(MetabaseStream):
+class Databases(MetabaseStream):
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        result = response.json().get("data", [])
+        yield from result
+
+    def path(self, **kwargs) -> str:
+        return "database"
 
-    response_entity = "data"
+
+class DatasetQueryViews(IncrementalMetabaseStream):
+    primary_key = None
+    cursor_field = "viewed_on"
+    http_method = "POST"
+    page_size = 10000
+    page_offset = 0
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        data_result = response.json().get("data", [])
+        rows_result = data_result.get("rows", [])
+        header = [
+            "viewed_on",
+            "card_id",
+            "card_name",
+            "query_hash",
+            "type",
+            "collection_id",
+            "collection",
+            "viewed_by_id",
+            "viewed_by",
+            "saved_by_id",
+            "saved_by",
+            "database_id",
+            "source_db",
+            "table_id",
+            "table",
+        ]
+        row_num = 0
+        for row in literal_eval(str(rows_result)):
```

Comment:
Could explain why you use `literal_eval` here? Isn't it dangerous to evaluate the content of an API response that can lead to injection?

Reply:
I did that because the API seems to return a JSON response with non-json array values:
```
{
  "data": { 
     "rows":[
        [ "value_0", "value_1", ... ]
     ]
   }
}
```

So it seems we have to parse the string value as a python array first